tensorflow/c/experimental/filesystem/plugins/gcs/gcs_filesystem.cc

```diff
@@ -410,6 +418,70 @@ void NewReadOnlyMemoryRegionFromFile(const TF_Filesystem* filesystem,
   }
 }
 
+void CreateDir(const TF_Filesystem* filesystem, const char* path,
+               TF_Status* status) {
+  std::string bucket, object;
+  ParseGCSPath(path, true, &bucket, &object, status);
+  if (TF_GetCode(status) != TF_OK) return;
+  auto gcs_file = static_cast<GCSFile*>(filesystem->plugin_filesystem);
+  if (object.empty()) {
+    auto bucket_metadata = gcs_file->gcs_client.GetBucketMetadata(bucket);
+    TF_SetStatusFromGCSStatus(bucket_metadata.status(), status);
+    return;
+  }
+
+  MaybeAppendSlash(&object);
```

Comment:
Nit: Can we ensure that `object` always ends with a `/`?

Reply:
Sorry I dont understand what you mean. `MaybeAppendSlack` will ensure object will end with a '/'

Reply:
I meant `object` is always produced by other functions. Can we ensure that those functions add the trailing slash?

Basically, from the point of view of a caller to `ParseGCSPath`, `object` either always has the trailing `/` or not but not something in between. Otherwise, you always have to write code that uses both `ParseGCSPath` and `MaybeAppendSlash`.

Reply:
I think we could force `object` always has the trailing `/` or not in `ParseGCSPath` but it causes many unwanted behaviors.

In the local filesystem, we have a function `mkdir` to create directory and the directory name can have `/` or not, it does not matter.

But with `gcs` filesystem, we have to create an empty object with the name ends with `/` so Google Cloud Console could display it as a directory.

If we force `object` does not have `/` at the end, Google Cloud Console will display that object as a empty object.
If we force `object` has `/` at the end, Google Cloud Console will display everything as a directory ( even the object's size is bigger than 0 ).

Reply:
Ok, then let's leave it as it is